src/Microsoft.Identity.Web.TokenAcquisition/ConfidentialClientApplicationBuilderExtension.cs

```diff
@@ -58,51 +90,47 @@ public static async Task<ConfidentialClientApplicationBuilder> WithClientCredent
                             else
                             {
                                 Logger.UsingManagedIdentity(logger);
-                                return builder.WithClientAssertion((credential.CachedValue as ManagedIdentityClientAssertion)!.GetSignedAssertionAsync);
+                                return credential;
                             }
                         }
                         if (credential.SourceType == CredentialSource.SignedAssertionFilePath)
                         {
                             if (!credential.Skip)
                             {
                                 Logger.UsingPodIdentityFile(logger, credential.SignedAssertionFileDiskPath ?? "not found");
-                                return builder.WithClientAssertion((credential.CachedValue as AzureIdentityForKubernetesClientAssertion)!.GetSignedAssertionAsync);
+                                return credential;
                             }
                         }
                         if (credential.SourceType == CredentialSource.SignedAssertionFromVault)
                         {
                             if (!credential.Skip)
                             {
                                 Logger.UsingSignedAssertionFromVault(logger, credential.KeyVaultUrl ?? "undefined");
-                                return builder.WithClientAssertion((credential.CachedValue as ClientAssertionProviderBase)!.GetSignedAssertionAsync);
+                                return credential;
                             }
                         }
                     }
 
                     if (credential.CredentialType == CredentialType.Certificate)
                     {
-                        if (credential.Certificate !=null)
+                        if (credential.Certificate != null)
                         {
                             Logger.UsingCertThumbprint(logger, credential.Certificate.Thumbprint);
-                            return builder.WithCertificate(credential.Certificate);
+                            return credential;
                         }
                     }
 
                     if (credential.CredentialType == CredentialType.Secret)
                     {
-                        return builder.WithClientSecret(credential.ClientSecret);
+                        return credential;
                     }
                 }
             }
 
-            if (clientCredentials.Any(c => c.CredentialType == CredentialType.Certificate))
-            {
-                throw new ArgumentException(
-                    IDWebErrorMessage.ClientCertificatesHaveExpiredOrCannotBeLoaded,
-                    nameof(clientCredentials));
-            }
+            throw new ArgumentException(
+               IDWebErrorMessage.ClientCertificatesHaveExpiredOrCannotBeLoaded + errorMessage,
+               nameof(clientCredentials));
```

Comment:
This also part of fix for #3000 - if none of the credentials can be loaded, we should fail with a meaningful message. Instead, the Exception here is conditioned on the existence of a certificate, so this code didn't execute for FIC.